oras/provider.py

```diff
@@ -233,11 +234,54 @@ def get_tags(
 
         :param container:  parsed container URI
         :type container: oras.container.Container or str
-        :param N: number of tags
+        :param N: number of tags, -1 for all
```

Comment:
-1 for all is something the endpoint honors?

Reply:
I completely missed that it would be sent to the endpoint haha. Apparently it works in the tests, but idk – should we make it `None` in the case of all tags and retrieve tags with N = "10k" or other random high number?

Reply:
I think as a user my expectation is to get back all tags - it would be annoying if I didn't! I think the easiest thing to do is just pass in the really high value for N, and if someone gives it None we would set to 0 and the distribution spec says a value of 0 should return an empty list. So something like:

```python
N = N or 0
```
I thought about some boolean like "retrieve_all" but then that could conflict with N and it gets too complicated, so I like just a high value of N. What do you think?